backend/internal/database/book.go

```diff
@@ -402,8 +402,9 @@ func (db *BookDatabase) FetchFavorites(ctx context.Context, userID string) ([]mo
  * Fetches a single favorited book for the user, if it is favorited
  */
 func (db *BookDatabase) FetchFavoritedBook(ctx context.Context,
-	userID string, bookID string) (string, error) {
+	userID string, bookID string) (bool, error) {
 
+	var favorited bool
 	var book_id string
 
 	query := "SELECT book_id FROM favorite_books JOIN books ON favorite_books.book_id = books.id " +
```

Comment:
try `SELECT count(*) > 0 FROM favorite_books WHERE user_id=$1 AND book_id=$2`, we shouldn't need to join here since we're just checking the ID. Also there's no need to return the ID, we can just do a check if the row exists with count(*) > 0